web/src/LogActions.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect } from "react"
+import { useStorageState } from "react-storage-hooks"
+import styled from "styled-components"
+import { incr } from "./analytics"
+import ClearLogs from "./ClearLogs"
+import { Color, FontSize, mixinResetButtonStyle } from "./style-helpers"
+
+export const LogFontSizeScaleLocalStorageKey = "tilt.global.log-font-scale"
+export const LogFontSizeScaleCSSProperty = "--log-font-scale"
+export const LogFontSizeScaleMinimumPercentage = 10
+
+const LogActionsGroup = styled.div`
+  margin-left: auto;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`
+
+const FontSizeControls = styled.div`
+  color: ${Color.gray6};
+  vertical-align: middle;
+  display: flex;
+  flex-wrap: nowrap;
+`
+
+const FontSizeControlsDivider = styled.div`
+  font-size: ${FontSize.default};
+  &:after {
+    content: "|";
+  }
+`
+
+const FontSizeButton = styled.button`
+  ${mixinResetButtonStyle}
+  color: ${Color.gray6};
+
+  padding: 0 4px;
+
+  &:after {
+    content: "A";
+  }
+`
+
+const FontSizeDecreaseButton = styled(FontSizeButton)`
+  font-size: ${FontSize.smallest};
+`
+FontSizeDecreaseButton.displayName = "LogFontSizeDecreaseButton"
+
+const FontSizeIncreaseButton = styled(FontSizeButton)`
+  font-size: ${FontSize.default};
+`
+FontSizeIncreaseButton.displayName = "LogFontSizeIncreaseButton"
+
+export const LogsFontSize: React.FC = () => {
+  // this uses `useStorageState` directly vs `usePersistentState` wrapper because it's a global setting
+  // (i.e. log zoom applies across all Tiltfiles)
+  const [logFontScale, setLogFontSize] = useStorageState<string>(
+    localStorage,
+    LogFontSizeScaleLocalStorageKey,
+    () =>
+      document.documentElement.style.getPropertyValue(
+        LogFontSizeScaleCSSProperty
+      )
+  )
+  useEffect(() => {
```

Comment:
if logFontScale is invalid when we read it, it seems like we'd want to fix that in the local variable here, rather than letting all the downstream functions do their own validation?

Reply:
Tweaked in b4c82591e5b107070a1112cd8ad7d8bc3057441b - will reset to `100%` here if invalid, so there's no need for extra validation when stepping.

(FWIW None of this _should_ be needed - I'm just extra paranoid because I've firsthand experienced lots of icky behaviors with localStorage, particularly when apps change "schema" of data in keys and havoc ensues)